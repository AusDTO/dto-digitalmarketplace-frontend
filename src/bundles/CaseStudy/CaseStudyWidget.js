import React from 'react'
import { Provider } from 'react-redux'
import Registry from '../../registry'

// Need to make this work.
// import { CaseStudy, createStore } from '../bundles/CaseStudy'
import createStore from './redux/create'
import CaseStudyForm from './components/CaseStudyForm'

const CaseStudyWidget = (props, _context) => {
  const store = createStore(props)
  return (
    <Provider store={store}>
      <CaseStudyForm />
    </Provider>
  )
}

Registry.add({
  casestudy: CaseStudyWidget
})

export default Registry
