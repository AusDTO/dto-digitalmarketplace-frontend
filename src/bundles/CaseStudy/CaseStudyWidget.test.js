// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import { CaseStudyWidget } from './CaseStudyWidget';
import renderer from 'react-test-renderer';

test('CaseStudyWidget renders without errrors', () => {
  const component = renderer.create(
    <CaseStudyWidget />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})