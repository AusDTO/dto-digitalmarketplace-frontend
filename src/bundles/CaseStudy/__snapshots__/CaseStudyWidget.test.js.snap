exports[`test CaseStudyWidget renders without errrors 1`] = `
<div
  className="layout">
  <form
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex="-1" />
  <form
    action={undefined}
    id="casestudy__create"
    method="post"
    onReset={[Function]}
    onSubmit={[Function]}>
    <div
      className="field">
      <label
        className="question-heading"
        htmlFor="title">
        Give your case study a title
      </label>
      <input
        className=""
        id="title"
        name="title"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="" />
    </div>
    <div
      className="field">
      <label
        className="question-heading"
        htmlFor="client">
        Who was the client?
      </label>
      <input
        className=""
        id="client"
        name="client"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="" />
    </div>
    <div
      className="field">
      <label
        className="question-heading"
        htmlFor="timeframe">
        What was the time frame?
      </label>
      <p
        className="hint">
        For example,  January 2016 â€” June 2016
      </p>
      <input
        className=""
        id="timeframe"
        name="timeframe"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="" />
    </div>
    <div
      className="field">
      <label
        htmlFor="opportunity">
        Outline the problem or opportunity
      </label>
      <p
        className="hint">
        Describe the project goal and any relevant background information.
      </p>
      <div>
        <textarea
          className=""
          defaultValue=""
          id="opportunity"
          name="opportunity"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]} />
        <span
          className="limit">
          200 words remaining
        </span>
      </div>
    </div>
    <div
      className="field">
      <label
        htmlFor="approach">
        Describe your approach
      </label>
      <p
        className="hint">
        How did your capabilities and methods contribute to achieving the project goals?
      </p>
      <div>
        <textarea
          className=""
          defaultValue=""
          id="approach"
          name="approach"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]} />
        <span
          className="limit">
          200 words remaining
        </span>
      </div>
    </div>
    <div
      className="field">
      <label
        htmlFor="outcome-0">
        What was the outcome?
      </label>
      <p
        className="hint">
        List the key benefits of this project.
      </p>
      <div>
        <div
          className="list-entry">
          <label
            className="text-box-number-label"
            htmlFor="outcome">
            1
          </label>
          <input
            className="text-box "
            defaultValue=""
            id="outcome"
            name="outcome[]"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text" />
        </div>
        <button
          className="button-secondary"
          onClick={[Function]}
          type="button">
          Add another row
        </button>
      </div>
    </div>
    <div
      className="field">
      <label
        htmlFor="projectLinks-0">
        Project links
      </label>
      <p
        className="hint">
        Link to any supporting material for your case study. This can include a case study on your  website, case study video or the live project.
      </p>
      <div>
        <div
          className="list-entry">
          <label
            className="text-box-number-label"
            htmlFor="projectLinks">
            1
          </label>
          <input
            className="text-box "
            defaultValue=""
            id="projectLinks"
            name="projectLinks[]"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text" />
        </div>
        <button
          className="button-secondary"
          onClick={[Function]}
          type="button">
          Add another row
        </button>
      </div>
    </div>
    <div
      className="field">
      <input
        checked={false}
        id="acknowledge"
        name="acknowledge"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        type="checkbox" />
      <label
        htmlFor="acknowledge">
        I acknowledge this case study may be shared with registered buyers in the Digital Marketplace.
      </label>
    </div>
    <input
      onClick={[Function]}
      role="button"
      type="submit"
      value="Submit" />
  </form>
</div>
`;
